prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -127,6 +119,9 @@ def __call__(self, a: str) -> float:
 class BuyTokens(MarketFunction):
     def __init__(self, market_type: MarketType, outcome: str):
         self.outcome = outcome
+        self.outcome_bool = get_boolean_outcome(
```

Comment:
get_boolean_outcome was being called a few times in the code, so moved it here to avoid repeated calls